python35_filter.cpp

```diff
@@ -324,7 +324,12 @@ bool Python35Filter::reconfigure(const string& newConfig)
 {
 	lock_guard<mutex> guard(m_configMutex);
 
-	// Cleanup Loaded module first
+	PyGILState_STATE state = PyGILState_Ensure(); // acquire GIL
+
+	// Reload Python module: get a new PyObject
+	PyObject* newModule = PyImport_ReloadModule(m_pModule);
```

Comment:
Need to check return value of PyImport_ReloadModule(), it may return NULL in some cases.

From https://docs.python.org/3/c-api/import.html#c.PyImport_ReloadModule:

PyObject* PyImport_ReloadModule(PyObject *m)
    Return value: New reference.

    Reload a module. Return a new reference to the reloaded module, or NULL with an exception set on failure (the module still exists in this case).

Reply:
Python35Filter::reconfigure
// Set reloaded module
m_pModule = newModule;

In Python35Filter::configure we have a check

        // 2) Import Python script if module object is not set
        if (!m_pModule)
        {
                m_pModule = PyImport_ImportModule(m_pythonScript.c_str());
        }

        // Check whether the Python module has been imported
        if (!m_pModule)
        {
                // Failure
         ....
       }
....

Reply:
It might be better to log failure while reloading module right away rather than letting that error propagate to another function, where we cannot make out that difference.